### ISSUE TITLE
[Android] No text on main menu

### ISSUE THREAD

**Issue by Wayward1** [minetest/minetest#2973]
After git cloning and building Minetest in a new directory, there is no text on any of the buttons on the main menu, as seen here:
![screenshot_2015-07-29-17-44-29](https://cloud.githubusercontent.com/assets/8821916/8970940/8deebc94-361a-11e5-9049-a5b6fcc63a86.png)
![screenshot_2015-07-29-17-53-28](https://cloud.githubusercontent.com/assets/8821916/8970976/cb24e8d6-361a-11e5-9353-9717b325dd2f.png)

**Comment by est31** [comment]
Well, this is most likely due to the recent iconv changes. I couldn't reproduce back when I tested it. I'll test it again.

**Comment by est31** [comment]
Tested it again, it works. Will do a clean re-clone.

**Comment by est31** [comment]
Here a screenshot of a fresh clone:
![screenshot](https://cloud.githubusercontent.com/assets/8872119/8973456/27815304-3665-11e5-9165-a8bce06ddcd1.png)

**Comment by Wayward1** [comment]
Weird... I'll try another re-clone.

**Comment by Wayward1** [comment]
I just re-cloned and built, and still got the same result. I'm not sure what else could be causing this...

**Comment by est31** [comment]
Can you share the logcat output directly after "TESTING UTF TRANSCODING FUNCTIONS", when using [this branch](https://github.com/est31/minetest/tree/wide_to_utf8_test)?

**Comment by est31** [comment]
If everything is right, it would look like this:
```
V/minetest(16505): 2015-07-30 21:37:57: ERROR[main]: TESTING UTF TRANSCODING FUNCTIONS....
E/minetest(16505): TESTING UTF TRANSCODING FUNCTIONS....
V/minetest(16505): 2015-07-30 21:37:57: ERROR[main]: 68656c6c6f
E/minetest(16505): 68656c6c6f
V/minetest(16505): 2015-07-30 21:37:57: ERROR[main]: 68656c6c6f
E/minetest(16505): 68656c6c6f
```

**Comment by Wayward1** [comment]
Here's the logcat output:

`V/minetest(15790): 2015-07-30 18:41:19: ERROR[main]: TESTING UTF TRANSCODING FUNCTIONS....
E/minetest(15790): TESTING UTF TRANSCODING FUNCTIONS....
V/minetest(15790): 2015-07-30 18:41:19: ERROR[main]: 010101010101
E/minetest(15790): 010101010101
V/minetest(15790): 2015-07-30 18:41:19: ERROR[main]: 68656c6c6f
E/minetest(15790): 68656c6c6f`

**Comment by est31** [comment]
As it seems, that you are experiencing #2782.

**Comment by Wayward1** [comment]
Does that mean it's a problem with my build, or a genuine issue in Minetest?

**Comment by est31** [comment]
Well, that's the question...
We'll need more people who do android builds to tell us whether it works for them or not.

**Comment by Wayward1** [comment]
Just tested on a fresh OS, same results.

**Comment by est31** [comment]
OK to find out whether its a build system issue, or a device issue, can you reproduce the bug for the latest [nightly apk](https://jenkins.unix-experience.fr/job/minetest-android-OfficialRepository-RollingRelease/158/artifact/build/android/bin/Minetest-release-unsigned.apk)?

**Comment by est31** [comment]
@Wayward1 do you use release mode or debug mode to compile? Can you try debug mode?

**Comment by Wayward1** [comment]
I tried both release and debug. Also, for some reason that apk won't install at all.

**Comment by est31** [comment]
Can you try [this apk](https://jenkins.unix-experience.fr/job/minetest-android-OfficialRepository-RollingRelease/160/artifact/build/android/bin/Minetest-debug.apk)? The problem should be solved, I've contacted nrz.

**Comment by Wayward1** [comment]
Ok, that one installed and it seems fine.

**Comment by est31** [comment]
Its the stable-0.4 branch though, so not much to judge, contacted nrz to fix that problem.

**Comment by est31** [comment]
@Wayward1 can you reproduce with this [apk](https://jenkins.unix-experience.fr/job/minetest-android-OfficialRepository-RollingRelease/lastBuild/artifact/build/android/bin/Minetest-debug-unaligned.apk)? Its the build of the latest commit bcf38a2ad19116b50d0456bdd6f79d0d218e39d1

**Comment by Wayward1** [comment]
Yep, same result.

**Comment by est31** [comment]
Ok, then this bug seems to be device specific, because I can't reproduce it with that apk on my device.

**Comment by Wayward1** [comment]
I don't think so... I just tested that apk on my brother's Samsung Galaxy S4 Mini, with the same result.

**Comment by est31** [comment]
"Fixed" with 457d42c5997cc430231ed221f06d8f4bba2100e7, re-opening #2782.
